analyzers/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -66,58 +66,58 @@ public static bool IsStaticObjectEquals(this IMethodSymbol methodSymbol)
                 && methodSymbol.MethodKind == MethodKind.Ordinary
                 && methodSymbol.Name == nameof(object.Equals)
                 && methodSymbol.IsInType(KnownType.System_Object)
-                && HasCorrectArguments()
+                && HasCorrectParameters()
                 && methodSymbol.ReturnType.Is(KnownType.System_Boolean);
 
-            bool HasCorrectArguments() =>
+            bool HasCorrectParameters() =>
                 methodSymbol.Parameters.Length == 2
                 && methodSymbol.Parameters[0].Type.Is(KnownType.System_Object)
                 && methodSymbol.Parameters[1].Type.Is(KnownType.System_Object);
         }
 
         public static bool IsObjectGetHashCode(this IMethodSymbol methodSymbol) =>
             methodSymbol != null
-            && (methodSymbol.IsOverride || methodSymbol.IsInType(KnownType.System_Object))
             && methodSymbol.MethodKind == MethodKind.Ordinary
             && methodSymbol.Name == nameof(object.GetHashCode)
+            && (methodSymbol.IsOverride || methodSymbol.IsInType(KnownType.System_Object))
             && methodSymbol.Parameters.Length == 0
             && methodSymbol.ReturnType.Is(KnownType.System_Int32);
 
         public static bool IsObjectToString(this IMethodSymbol methodSymbol) =>
             methodSymbol != null
-            && (methodSymbol.IsOverride || methodSymbol.IsInType(KnownType.System_Object))
             && methodSymbol.MethodKind == MethodKind.Ordinary
             && methodSymbol.Name == nameof(object.ToString)
+            && (methodSymbol.IsOverride || methodSymbol.IsInType(KnownType.System_Object))
             && methodSymbol.Parameters.Length == 0
             && methodSymbol.ReturnType.Is(KnownType.System_String);
 
         public static bool IsIDisposableDispose(this IMethodSymbol methodSymbol)
         {
             const string explicitName = "System.IDisposable.Dispose";
             return methodSymbol != null
+                && (methodSymbol.Name == nameof(IDisposable.Dispose) || methodSymbol.Name == explicitName)
```

Comment:
I'm not sure this change makes sense - checking a bool or int property should be less expensive than checking the name

(I guess you can make this change in a different PR if you want)

Reply:
(discussed offline - the changes were made to keep a common checking pattern  - first name, then params)